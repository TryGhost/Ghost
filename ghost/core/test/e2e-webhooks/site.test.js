const {agentProvider, mockManager, fixtureManager, matchers} = require('../utils/e2e-framework');
const {anyGhostAgent, anyContentVersion, anyNumber} = matchers;

describe('site.* events', function () {
    let adminAPIAgent;
    let webhookMockReceiver;

    before(async function () {
        adminAPIAgent = await agentProvider.getAdminAPIAgent();
        await fixtureManager.init('integrations');
        await adminAPIAgent.loginAsOwner();
    });

    beforeEach(function () {
        webhookMockReceiver = mockManager.mockWebhookRequests();
    });

    afterEach(function () {
        mockManager.restore();
    });

    it('site.changed event is triggered', async function () {
        const webhookURL = 'https://test-webhook-receiver.com/site-changed';
        await webhookMockReceiver.mock(webhookURL);
        await fixtureManager.insertWebhook({
            event: 'site.changed',
            url: webhookURL
        });

        await adminAPIAgent
            .post('posts/')
            .body({
                posts: [{
                    title: 'webhookz',
                    status: 'published',
                    mobiledoc: fixtureManager.get('posts', 1).mobiledoc
                }]
            })
            .expectStatus(201);

        await webhookMockReceiver.receivedRequest();

        webhookMockReceiver
            .matchHeaderSnapshot({
                'content-version': anyContentVersion,
                'content-length': anyNumber,
                'user-agent': anyGhostAgent
            })
            .matchBodySnapshot();
    });
});
