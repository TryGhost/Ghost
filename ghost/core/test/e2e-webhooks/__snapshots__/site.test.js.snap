// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site.* events site.changed event is triggered 1: [headers] 1`] = `
Object {
  "accept-encoding": "gzip, deflate, br",
  "content-length": Any<Number>,
  "content-type": "application/json",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "user-agent": StringMatching /Ghost\\\\/\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\\\s\\\\\\(https:\\\\/\\\\/github\\.com\\\\/TryGhost\\\\/Ghost\\\\\\)/,
}
`;

exports[`site.* events site.changed event is triggered 2: [body] 1`] = `Object {}`;
